design/design.md

```diff
@@ -0,0 +1,414 @@
+multisignature scheme, also called Mithril, that lets stakeholders collectively
+sign a message that convinces a verifier that nodes with more than some
+threshold of the participating stake signed this message.
+To faciliate fast node bootstrapping, this message is a commitment to the whole
+state of the ledger.
+ALBA can be used to construct a new weighted threshold multisignature scheme
+with a significantly better proof size which
+can allow Cardano nodes to process the chain of Mithril certificates faster.
+If a SNARK is used to compress Mithril certificates to a very small size,
+ALBA can reduce the overall proving time.
+
+Peras and Leios are new extensions of the Ouroboros consensus protocol used in
+Cardano.
+Ouroboros Peras aims to reduce the block settlement time by executing a byzantine fault
+tolerant agreement protocol to decide the next block when adversarial
+conditions are mild.
+The consensus participants certify the chosen block to allow bootstrapping
+nodes to follow the right chain.
+A natural method for this certification is to use a
+weighted threshold multisignature scheme constructed using decentralized ALBA.
+
+Ouroboros Leios aims to increase the throughput of the blockchain in terms of
+the number of transactions per second.
+It does so by decoupling the block payloads from the consensus mechanism which
+operates only on references to blocks.
+To make sure that the chosen block's payload is correct and available
+(which potentially could be not due to malicious behavior),
+all blocks in the input to the consensus protocol need to have
+a certificate of availability proving that at least one honest node has verified
+the block content and is available to distribute it.
+A natural solution to certify data availability would be to use a
+weighted threshold multisignature scheme constructed using decentralized ALBA.
+
+Cross-chain bridges are protocols allowing one blockchain to verify
+information on another blockchain.
+In essense this is the same problem that the Mithril solves with a potentially
+added constraint that on-chain verification must be computationally cheap.
+Whereas Mithril is specific to the Cardano blockchain, one can envision
+a similar protocol, powered by ALBA, for other cryptocurrencies including
+partner chains like Midnight.
+
+Applications of ALBA are not limited to Cardano and blockchains in general.
+For instance, the [paper](https://eprint.iacr.org/2023/1655) describes how
+this primitive can be used to optimize the prover running time of
+a Universally Composable SNARK.
+
+## Requirements
+
+We intend to make an implementation of ALBA that can be used in
+Mithril, Peras, Leios, cross-chain bridges and potentially other protocols
+inside and outside the Cardano universe.
+Our implementation will mostly be guided by our main "clients":
+Mirthril, Peras and Leios.
+
+ALBA is a generic tool that lets one prove knowledge of any kind of data
```

Comment:
it's knowledge of a _bound_, be precise.

Reply:
Wouldn't this be _how_ your primitive is designed to solve the prof of lower bound rather than the goal of the primitive though?